File: src/combined/Models/Entities/Link.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Transactions;

namespace v2.Models.Entities;

public class Link
{
    public long Id { get; init; }
    public long EventID { get; init; }
    public long PersonID { get; init; }
    public DateTime Date { get; init; }
    public string LinkURL { get; init; }
    public string LinkGUID { get; init; }
}

File: src/combined/ClientApp/src/JsModels/link.ts
export default class Link {
    id?: number;
    eventID: number;
    personID?: number;
    date: Date;
    linkUrl: string;
    linkGUID: string;

    constructor(
        eventID: number,
        date: Date,
        linkUrl: string,
        linkGUID: string,
        personID?: number,
        id?: number

    ) {
        this.eventID = eventID;
        this.personID = personID ?? undefined;
        this.date = date ?? new Date();
        this.linkUrl = linkUrl;
        this.linkGUID = linkGUID;
        this.id = undefined;
    }
}